include/lsst/afw/image/Pixel.h

```diff
@@ -233,6 +235,13 @@ class Pixel : public detail::MaskedImagePixel_tag {
         return !(lhs == rhs);
     }
 
+    /// Return a hash of this object.
+    std::size_t hash_value() const noexcept {
+        // Convert to double to avoid inconsistently hashing equal numbers of different types
```

Comment:
Are you trying to make image-like objects with different pixel types but the same values hash as equal?  I wouldn't have thought we had a use case for cross-type comparisons, but even if we do I'm not sure it's desirable (is that what our `operator==` does?!).

Reply:
Yup: https://github.com/lsst/afw/blob/master/include/lsst/afw/image/Pixel.h#L224-L228. This one actually kind of makes sense, especially if you're doing the comparison on the Python side -- programmers won't think of `Pixel<int>` and `Pixel<double>` as distinct types there.

In any case, one of the few hard rules for hash functions is that equal objects must have equal hashes, and apparently `4 == 4.0`. I've got a test case that covers this; we can remove it if we change the behavior of `operator==`.